src/core/hooking/Hooking.cpp

```diff
@@ -5,17 +5,18 @@
 #include "VMTHook.hpp"
 #include "game/hooks/Hooks.hpp"
 #include "game/pointers/Pointers.hpp"
+#include "../memory/ModuleMgr.hpp"
 
 namespace YimMenu
 {
 	Hooking::Hooking()
 	{
-		BaseHook::Add<Window::WndProc>(new DetourHook("WndProc", Pointers.WndProc, Window::WndProc));
+		Window::OriginalWndProc = reinterpret_cast<WNDPROC>(SetWindowLongPtrW(Pointers.Hwnd, GWLP_WNDPROC, reinterpret_cast<LONG_PTR>(Window::WndProc)));
+	
+		BaseHook::Add<Window::SetCursorPos>(new DetourHook("SetCursorPos", ModuleMgr.Get("user32.dll")->GetExport<void*>("SetCursorPos"), Window::SetCursorPos));
 
-		auto swapchain = new VMTHook<SwapChain::VMTSize>("SwapChain", *Pointers.SwapChain);
-		swapchain->Hook(SwapChain::VMTPresentIdx, SwapChain::Present);
-		swapchain->Hook(SwapChain::VMTResizeBuffersIdx, SwapChain::ResizeBuffers);
-		BaseHook::Add<SwapChain::Present>(swapchain);
+		BaseHook::Add<SwapChain::Present>(new DetourHook("SwapChain::Present", GetVF(*Pointers.SwapChain, SwapChain::VMTPresentIdx), SwapChain::Present));
+		BaseHook::Add<SwapChain::ResizeBuffers>(new DetourHook("SwapChain::ResizeBuffers", GetVF(*Pointers.SwapChain, SwapChain::VMTResizeBuffersIdx), SwapChain::ResizeBuffers));
```

Comment:
Any reason for not making use of the existing VMT hook impl?

Reply:
RDR2 would always crash when using VMT hooks. No idea why,  but I just did what works.

Reply:
I'll attempt to use it again

Reply:
Nope, didn't work for me. Left a comment saying TODO